packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1189,6 +1204,32 @@ abstract class ResidentRunner extends ResidentHandlers {
     globals.printTrace('complete');
   }
 
+  Future<DartBuildResult> runDartBuild() async {
+    globals.printTrace('runDartBuild()');
+    if (_dartBuildResult != null && _dartBuildResult!.isBuildUpToDate(globals.fs)) {
+      globals.printTrace('runDartBuild() - up-to-date already');
+      return _dartBuildResult!;
+    }
+    globals.printTrace('runDartBuild() - will perform dart build');
+
+    final BuildResult lastBuild = await globals.buildSystem.build(
+      const DartBuildForWeb(<Dart2WebTarget>[]),
```

Comment:
@mkustermann is this introducing an additional build?

Reply:
So this code here (`resident_runner.dart`) is used in hot-reload/hot-restart scenarios.

We have a check above `_dartBuildResult!.isBuildUpToDate()` that will return early if nothing changed. But if any of the hook dependencies changed, we may need to re-run the dart build, which we do here.

@mosuem Did you mean something else?

Reply:
Resolved offline - This is an additional build, but only of the `DartBuildForWeb` target, which is necessary to update the changed assets.